### ISSUE TITLE
New Account 

### ISSUE THREAD

**Issue by DHAIRYA2311** [louislam/uptime-kuma#1492]
### ⚠️ Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### 📝 Describe your problem

In Uptimer Kuma how can we sign up there is no button in website after running it

### 🐻 Uptime-Kuma Version

1.11.3

### 💻 Operating System and Arch

Mac OS

### 🌐 Browser

Chrome

**Comment by simon-lorenz** [comment]
When visiting the website for the first time you should be automatically redirected to the `/setup` path.

I just tested this with a fresh 1.11.3 installation (via docker):

![image](https://user-images.githubusercontent.com/34041551/163347408-4ba0d760-6845-4c86-963b-796c10f8ce26.png)

**Comment by louislam** [comment]
Why use old version?
If you are a new user, please follow the installation guide in this repo.

**Comment by crs369** [comment]
Hello,

this is my first contact with the uptime-kuma community.

Thanks for the great uptime-kuma.

I have running uptime-kuma quite a while without problems, but a few day ago it brokes.

I think this is related with the update procuedure that i using daily;

docker pull louislam/uptime-kuma:latest
docker stop uptime-kuma
docker rm uptime-kuma
docker run -d --restart=always -p 80:3001 -v uptime-kuma:/app/data --name uptime-kuma louislam/uptime-kuma:latest
docker image prune -f

For now i am getting the same dialog above and not the usual dashboard.

I have playback my backup and all went fine - update -> broke.

Any hints?

Thanks
Rainer

**Comment by louislam** [comment]
What is your previous version?

> For now i am getting the same dialog above

You mean setup page?

**Comment by crs369** [comment]
I'm running Version: 1.13.2

Normally I'm getting the Status-Page http://kuma/status/default

After the update I'm getting the Page above.

**Comment by louislam** [comment]
Still not quite sure how is your context. 

Usually, people forget to use their previous volume, so the setup page will be shown.

Make sure you understand what is docker volume and try to find original volume name (Some tutorials or videos teach you binds to a folder)

**Comment by crs369** [comment]
docker volume is there and looking in the container i can access the data - looks fine for me.

I see in the error.log following output:

```
[2022-04-15 10:31:44] Error: ENOENT: no such file or directory, copyfile './data/kuma.db.bak20220415103141' -> './data/kuma.db'
    at Object.copyFileSync (node:fs:2804:3)
    at Function.restore (/app/server/database.js:453:16)
    at Function.patch2 (/app/server/database.js:220:18)
    at async Function.patch (/app/server/database.js:185:9)
    at async initDatabase (/app/server/server.js:1538:5)
    at async /app/server/server.js:211:5 {
  errno: -2,
  syscall: 'copyfile',
  code: 'ENOENT',
  path: './data/kuma.db.bak20220415103141',
  dest: './data/kuma.db'
}
```

My data in the volume:
```
root@kuma:/var/lib/docker/volumes/uptime-kuma/_data# ls -l
insgesamt 1449440

-rw-r--r-- 1 root root      12566 Apr 15 12:31 error.log
-rwxr-xr-x 1 root root      61440 Apr 15 12:31 kuma.db
-rwxr-xr-x 1 root root      61440 Apr 15 12:31 kuma.db.bak0
-rwxr-xr-x 1 root root    4083712 Okt 31 10:17 kuma.db.bak20211031091708
-rwxr-xr-x 1 root root  316579840 Dez  9 04:06 kuma.db.bak20211209030617
-rwxr-xr-x 1 root root 1162735616 Mär 26 04:06 kuma.db.bak20220326030617
-rwxr-xr-x 1 root root      32768 Apr 15 12:31 kuma.db-shm
-rwxr-xr-x 1 root root      32768 Apr 15 12:31 kuma.db-shm.bak0
-rwxr-xr-x 1 root root      32768 Okt 31 10:17 kuma.db-shm.bak20211031091708
-rwxr-xr-x 1 root root      32768 Dez  9 04:06 kuma.db-shm.bak20211209030617
-rwxr-xr-x 1 root root      32768 Mär 26 04:06 kuma.db-shm.bak20220326030617
-rwxr-xr-x 1 root root     494432 Apr 15 12:31 kuma.db-wal
-rwxr-xr-x 1 root root          0 Apr 15 12:31 kuma.db-wal.bak0
-rwxr-xr-x 1 root root          0 Okt 31 10:17 kuma.db-wal.bak20211031091708
-rwxr-xr-x 1 root root          0 Dez  9 04:06 kuma.db-wal.bak20211209030617
-rwxr-xr-x 1 root root          0 Mär 26 04:06 kuma.db-wal.bak20220326030617
drwxr-xr-x 2 root root       4096 Jan  4 16:40 upload
root@kuma:/var/lib/docker/volumes/uptime-kuma/_data#

```



****

**Comment by simon-lorenz** [comment]
Based on your logs it seems that some database patches failed (there should be additional log output above) and the backup restoration failed, because `kuma.db.bak20220415103141` does not exist.

Looking at the [code](https://github.com/louislam/uptime-kuma/blob/0c40e32d7572c670e88120828ae0be2184a7430e/server/database.js#L453) it would be interesting to know where `this.backupPath` comes from and why it references a file that does not exist.

You should probably open a separate issue for this and provide additional log output, if possible.

**Comment by louislam** [comment]
Oh no, I suggested that you should start your container in foreground (widthout `-d`), and then post the console output here. I think the database upgrade process is failed.

**Comment by crs369** [comment]
Ok. with the additional log output i can track the problem down. 
-> No Space left on device

For a first test i delete the file kuma.db.bak20220326030617 and the update progress does work.

Question: How to deal with this the right way?
1. Delete the files with \*bak\*?
2. Increase the storage space on disk?

Possibility i miss something - is there some option, so that i can give a maximal storage size, for example like prometheus?



```
root@kuma:/var/lib/docker/volumes/uptime-kuma/_data# ls -l
insgesamt 2724924
-rw-r--r-- 1 root root      11476 Apr  2 05:14 error.log
-rwxr-xr-x 1 root root 1302478848 Apr 11 21:55 kuma.db
-rwxr-xr-x 1 root root      61440 Okt 30 21:57 kuma.db.bak0
-rwxr-xr-x 1 root root    4083712 Okt 31 10:17 kuma.db.bak20211031091708
-rwxr-xr-x 1 root root  316579840 Dez  9 04:06 kuma.db.bak20211209030617
-rwxr-xr-x 1 root root 1162735616 Mär 26 04:06 kuma.db.bak20220326030617
-rwxr-xr-x 1 root root      32768 Apr 15 16:02 kuma.db-shm
-rwxr-xr-x 1 root root      32768 Okt 30 21:57 kuma.db-shm.bak0
-rwxr-xr-x 1 root root      32768 Okt 31 10:17 kuma.db-shm.bak20211031091708
-rwxr-xr-x 1 root root      32768 Dez  9 04:06 kuma.db-shm.bak20211209030617
-rwxr-xr-x 1 root root      32768 Mär 26 04:06 kuma.db-shm.bak20220326030617
-rwxr-xr-x 1 root root    4165352 Apr 15 16:02 kuma.db-wal
-rwxr-xr-x 1 root root          0 Okt 30 21:57 kuma.db-wal.bak0
-rwxr-xr-x 1 root root          0 Okt 31 10:17 kuma.db-wal.bak20211031091708
-rwxr-xr-x 1 root root          0 Dez  9 04:06 kuma.db-wal.bak20211209030617
-rwxr-xr-x 1 root root          0 Mär 26 04:06 kuma.db-wal.bak20220326030617
drwxr-xr-x 2 root root       4096 Jan  4 16:40 upload
root@kuma:/var/lib/docker/volumes/uptime-kuma/_data#

root@kuma:/var/lib/docker/volumes/uptime-kuma/_data# df -k .
Dateisystem                         1K-Blöcke Benutzt Verfügbar Verw% Eingehängt auf
/dev/mapper/vgubuntu2004server-root   8747728 8089960    193692   98% /


root@kuma:/var/lib/docker/volumes/uptime-kuma/_data# docker pull louislam/uptime-kuma:latest
latest: Pulling from louislam/uptime-kuma
f003217c5aae: Pull complete
e4c439747d16: Pull complete
28da4e420fe1: Pull complete
1c9c1c76f271: Pull complete
9c3fa31ad448: Pull complete
6f1f6d87adeb: Pull complete
95512224c4e9: Pull complete
bb3c76fb4e1e: Pull complete
486f3f3abd0e: Pull complete
1d68cf062130: Pull complete
Digest: sha256:761eb7718efdb5156380fe8fdc68f64174833647435f09bb4cc5f6f03cc43482
Status: Downloaded newer image for louislam/uptime-kuma:latest
docker.io/louislam/uptime-kuma:latest
root@kuma:/var/lib/docker/volumes/uptime-kuma/_data# docker stop uptime-kuma
uptime-kuma
root@kuma:/var/lib/docker/volumes/uptime-kuma/_data# docker rm uptime-kuma
uptime-kuma
root@kuma:/var/lib/docker/volumes/uptime-kuma/_data# docker run --restart=always -p 80:3001 -v uptime-kuma:/app/data --name uptime-kuma louislam/uptime-kuma:latest
==> Performing startup jobs and maintenance tasks
==> Starting application with user 0 group 0
Welcome to Uptime Kuma
Your Node.js version: 16
Node Env: production
Importing Node libraries
Importing 3rd-party libraries
Importing this project modules
Prepare Notification Providers
Version: 1.14.0
Creating express and socket.io instance
Server Type: HTTP
Data Dir: ./data/
Connecting to the Database
SQLite config:
[ { journal_mode: 'wal' } ]
[ { cache_size: -12000 } ]
SQLite Version: 3.36.0
Connected
Your database version: 10
Latest database version: 10
Database patch not needed
Database Patch 2.0 Process
patch-proxy.sql is not patched
Backing up the database
Closing the database
SQLite closed
Error: ENOSPC: no space left on device, copyfile './data/kuma.db' -> './data/kuma.db.bak20220415140902'
    at Object.copyFileSync (node:fs:2804:3)
    at Function.backup (/app/server/database.js:408:16)
    at Function.patch2Recursion (/app/server/database.js:320:18)
    at Function.patch2 (/app/server/database.js:206:28)
    at async Function.patch (/app/server/database.js:185:9)
    at async initDatabase (/app/server/server.js:1538:5)
    at async /app/server/server.js:211:5 {
  errno: -28,
  syscall: 'copyfile',
  code: 'ENOSPC',
  path: './data/kuma.db',
  dest: './data/kuma.db.bak20220415140902'
}
Start Uptime-Kuma failed due to issue patching the database
Please submit the bug report if you still encounter the problem after restart: https://github.com/louislam/uptime-kuma/issues
Patching the database failed!!! Restoring the backup
Trace: Error: ENOENT: no such file or directory, copyfile './data/kuma.db.bak20220415140902' -> './data/kuma.db'
    at Object.copyFileSync (node:fs:2804:3)
    at Function.restore (/app/server/database.js:453:16)
    at Function.patch2 (/app/server/database.js:220:18)
    at async Function.patch (/app/server/database.js:185:9)
    at async initDatabase (/app/server/server.js:1538:5)
    at async /app/server/server.js:211:5 {
  errno: -2,
  syscall: 'copyfile',
  code: 'ENOENT',
  path: './data/kuma.db.bak20220415140902',
  dest: './data/kuma.db'
}
    at process.<anonymous> (/app/server/server.js:1650:13)
    at process.emit (node:events:526:28)
    at emit (node:internal/process/promises:140:20)
    at processPromiseRejections (node:internal/process/promises:274:27)
    at processTicksAndRejections (node:internal/process/task_queues:97:32)
If you keep encountering errors, please report to https://github.com/louislam/uptime-kuma/issues

root@kuma:/var/lib/docker/volumes/uptime-kuma/_data# ls -l
insgesamt 1449436
-rw-r--r-- 1 root root      12046 Apr 15 16:09 error.log
-rwxr-xr-x 1 root root      61440 Apr 15 16:09 kuma.db
-rwxr-xr-x 1 root root      61440 Apr 15 16:09 kuma.db.bak0
-rwxr-xr-x 1 root root    4083712 Okt 31 10:17 kuma.db.bak20211031091708
-rwxr-xr-x 1 root root  316579840 Dez  9 04:06 kuma.db.bak20211209030617
-rwxr-xr-x 1 root root 1162735616 Mär 26 04:06 kuma.db.bak20220326030617
-rwxr-xr-x 1 root root      32768 Apr 15 16:09 kuma.db-shm
-rwxr-xr-x 1 root root      32768 Apr 15 16:09 kuma.db-shm.bak0
-rwxr-xr-x 1 root root      32768 Okt 31 10:17 kuma.db-shm.bak20211031091708
-rwxr-xr-x 1 root root      32768 Dez  9 04:06 kuma.db-shm.bak20211209030617
-rwxr-xr-x 1 root root      32768 Mär 26 04:06 kuma.db-shm.bak20220326030617
-rwxr-xr-x 1 root root     494432 Apr 15 16:09 kuma.db-wal
-rwxr-xr-x 1 root root          0 Apr 15 16:09 kuma.db-wal.bak0
-rwxr-xr-x 1 root root          0 Okt 31 10:17 kuma.db-wal.bak20211031091708
-rwxr-xr-x 1 root root          0 Dez  9 04:06 kuma.db-wal.bak20211209030617
-rwxr-xr-x 1 root root          0 Mär 26 04:06 kuma.db-wal.bak20220326030617
drwxr-xr-x 2 root root       4096 Jan  4 16:40 upload

root@kuma:/var/lib/docker/volumes/uptime-kuma/_data# df -k .
Dateisystem                         1K-Blöcke Benutzt Verfügbar Verw% Eingehängt auf
/dev/mapper/vgubuntu2004server-root   8747728 7225024   1058628   88% /
root@kuma:/var/lib/docker/volumes/uptime-kuma/_data#
```

**Comment by louislam** [comment]
> No Space left on device

Yes, it is the main reason.

I think your system disk has 8GB only which is too small. I suggest that you should backup the disk and find a way to increase the disk capacity to more than 30GB.

**Comment by mway-niels** [comment]
We ran into a similar issue after we upgraded from 1.11.4 to 1.15.1. We were redirected to the setup page and found the following error message in the log:

```
{"log":"Error: ENOSPC: no space left on device, copyfile './data/kuma.db' -\u003e './data/kuma.db.bak20220505070752'\n","stream":"stderr","time":"2022-05-05T07:08:08.026591073Z"}
```

We also don't see an obvious way to restore the previous (pre-upgrade) data. In my opinion it would be best if the user gets an error message in the UI informing him that the backup couldn't be completed and he has to revert the application version, increase storage size and try again instead of just redirecting to the setup page.
